### PR TITLE
session change trigger fix

### PR DESCRIPTION
**Pull request by talhadaar**
As pointed out by the question on [Substrate Polkadot Stackexchange](https://substrate.stackexchange.com/questions/2678/local-network-does-not-change-validator-set), the validator set doesn't change. This change fixes that issue and allows election to continue on as it should.

### CODE REVIEW

**Comment by EclesioMeloJunior** [comment]
@talhadaar thank you for being proactive and solve the problem described in the substrate stack exchange! 🎉

**Comment by talhadaar** [comment]
> @talhadaar thank you for being proactive and solve the problem described in the substrate stack exchange! 🎉

Thanks for asking the question!

**Review by qdm12 — APPROVED** [review]

**Review by EclesioMeloJunior — APPROVED** [review]
LGTM

**Comment by danforbes** [comment]
@talhadaar thank you so much for the assist 💪🏻 With your changes, I am able to add a new validator to the staking pool! Unfortunately, the newly added validator does not produce any blocks and the network still is not finalizing blocks. I don't recognize the first element in the list of new GRANDPA authorities - I would expect it to be `//Dave`. The other two authorities are `//Alice` and `//Bob`, which were set in the genesis block.
image

File: Cargo.lock [hunk 1]
```diff
@@ -4075,6 +4075,29 @@ dependencies = [
...
+ "sp-arithmetic",
+ "sp-core",
+ "sp-io",
+ "sp-npos-elections",
+ "sp-runtime",
+ "sp-std",
+ "static_assertions",
+ "strum",
+]
+
```
Comment (danforbes) [inline Cargo.lock]:
```suggestion
```

**Comment by talhadaar** [comment]
> @talhadaar thank you so much for the assist 💪🏻 With your changes, I am able to add a new validator to the staking pool! Unfortunately, the newly added validator does not produce any blocks and the network still is not finalizing blocks. I don't recognize the first element in the list of new GRANDPA authorities - I would expect it to be `//Dave`. The other two authorities are `//Alice` and `//Bob`, which were set in the genesis block. image

All of the 3 initial validators need to fired up and authorized for blocks to begin finalizing. I believe Charlie is not authorized on genesis, only Alice and Bob. It would be easier if you simply delete Charlie from initial validators, that should get the block finalization started.

**Comment by talhadaar** [comment]
> @talhadaar thank you so much for the assist 💪🏻 With your changes, I am able to add a new validator to the staking pool! Unfortunately, the newly added validator does not produce any blocks and the network still is not finalizing blocks. I don't recognize the first element in the list of new GRANDPA authorities - I would expect it to be `//Dave`. The other two authorities are `//Alice` and `//Bob`, which were set in the genesis block. image

The third entity is //Charlie i believe. Finalization will only begin once all genesis validators are running. Running Alice, Bob and Charlie will get your chain going. Then you can proceed to add more validators.

**Pull request merged**